examples/ex5p.cpp

```diff
@@ -260,6 +260,9 @@ int main(int argc, char *argv[])
    }
    else
    {
+      HypreParMatrix matB;
+      bVarf->FormRectangularSystemMatrix(empty_tdof_list, empty_tdof_list, matB);
```

Comment:
The segfault occurs on line 267 where `bVarf` is assembled after it is assembled on this line.  The segfault can be bypassed with the following steps:
- replace line 263 with `B = new HypreParMatrix();`
- replace line 264 with `bVarf->FormRectangularSystemMatrix(empty_tdof_list, empty_tdof_list, *B);`
- remove line 267
- change line 264 to `